### ISSUE TITLE
Looks like zodios is dead. What about ts-rest?

### ISSUE THREAD

**Issue by nd0ut** [astahmer/openapi-zod-client#305]
First of all, I want to express my gratitude for this amazing library!

I've noticed that there haven't been any new releases for Zodios for a while, and it seems like the project might be abandoned. As a result, I've started using ts-rest for my projects.

I'm curious if there are any plans to add built-in support for ts-rest? While I'm currently managing to meet my needs with custom templates, it would be fantastic to have an out-of-the-box template available.

Thanks for considering this!

**Comment by alden12** [comment]
This [openapi-ts-rest](https://github.com/Carminepo2/openapi-ts-rest?tab=readme-ov-file) package is still in development, but I am hoping this will be the package we are looking for once it's ready 🙏

**Comment by soulchild** [comment]
Zodios is dead in the water. Its maintainer seems to have vanished from the face of the earth. I've recently migrated to ts-rest which fell suspiciously silent soon after I finished the migration. I've asked the maintainers about the [future of the project](https://github.com/ts-rest/ts-rest/issues/797).

**Comment by astahmer** [comment]
you might be interested in https://github.com/astahmer/typed-openapi
but I don't really care about runtime validation with this, and you have to bring your own fetcher so more work for the users (but full customization without needing 50 CLI options 😅!)

**Comment by soulchild** [comment]
Thanks for the suggestion. But I'm really more interested in having a framework which gives me type-safe routing for my backend service. Your library "just" gives me a type-safe client library, right?

**Comment by astahmer** [comment]
ah yes you're right! then you might be interested in oRPC
https://orpc.unnoq.com/docs/comparison#comparison

**Comment by soulchild** [comment]
Yeah, I stumbled upon oRPC the other day as well. Looks very promising. But I don't feel like migrating away from ts-rest because we just migrated over from Zodios. 

Sadly, there seems to be only a handful of TS developers looking for a code-first approach to an OpenAPI backend. Which is weird because having everything in code (as opposed to wielding OpenAPI YAML) is an awesome experience, IMHO. Too bad that none of the libraries get enough traction. Everybody seems to be using tRPC, but their OpenAPI plugin was deprecated as well.

**Comment by astahmer** [comment]
yeah; agree it's very weird. 
recently I was using `hono` + `@hono/zod-openapi` but the typings are horribly slow, the DX is a nightmare with 200+ endpoints (that's why we just migrated to typed-openapi, it's still slow in the routers but at least the client is usable now)

**Comment by soulchild** [comment]
I only played around with `hono` and the `@hono/zod-openapi` extension. It also looked promising, but good to know that you were seeing such dramatic performance bottlenecks. Guess I won't be using it for anything substantial in the near future then… 😕 

Zodios and ts-rest also suffer from noticeably poorer TS performance (with about 80 endpoints). Desperately looking forward to the Golang rewrite of TS.

**Comment by FDiskas** [comment]
I would like to suggest 
openapi-fetch | 6 kB | 300k ops/s [example](https://github.com/vintasoftware/nextjs-fastapi-template/)
-- | -- | --
openapi-typescript-fetch | 3 kB | 300k ops/s (fastest)

**Comment by gunzip** [comment]
@FDiskas openapi-typescript is a good project if you don't need runtime validation and you trust the backend to always serve compliant requests (in all other cases typings won't reflect payload). I ended up writing my openapi-to-zod alternative in the lack of good spec-to-typescript generators.

**Comment by williamrjribeiro** [comment]
The most promising alternative I've found so far is [Platformatic's Massimo](https://blog.platformatic.dev/introducing-massimo). It doesn't rely on Zod, but it says it supports response validation against schema by using the `--validate-response` flag.

**Comment by gunzip** [comment]
Massimo is an excellent tool. The response validation happens in the Node.js client using AJV, which is very solid. However, I haven't been able to produce a _frontend_ client with runtime validation (it seems to be missing). I've experienced some other [shortcomings](https://gunzip.github.io/apical-ts/docs/comparison-with-alternative-libraries/#massimo) that prevented me from using it at the moment.
